src/components/side_panel/find_and_replace/find_and_replace_store.ts

```diff
@@ -307,18 +312,57 @@ export class FindAndReplaceStore extends SpreadsheetStore implements HighlightPr
       matches: [this.searchMatches[this.selectedMatchIndex]],
       searchOptions: this.searchOptions,
     });
+
+    if (this.irreplaceableMatchCount > 0) {
+      this.showReplaceWarningMessage(1, this.irreplaceableMatchCount);
+    }
+    this.irreplaceableMatchCount = 0;
+
     this.selectNextCell(Direction.next);
   }
   /**
    * Apply the replace function to all the matches one time.
    */
   replaceAll() {
+    const searchMatches = this.searchMatches;
+
     this.model.dispatch("REPLACE_SEARCH", {
       searchString: this.toSearch,
       replaceWith: this.toReplace,
-      matches: this.searchMatches,
+      matches: searchMatches,
       searchOptions: this.searchOptions,
     });
+
+    if (this.irreplaceableMatchCount > 0) {
+      this.showReplaceWarningMessage(searchMatches.length, this.irreplaceableMatchCount);
+    }
+    this.irreplaceableMatchCount = 0;
+  }
+
+  /**
+   * Show a warning message based on the number of matches replaced and irreplaceable.
+   */
+  private showReplaceWarningMessage(totalMatches: number, irreplaceableMatches: number): void {
+    const replaceableMatches = totalMatches - irreplaceableMatches;
+
+    if (replaceableMatches === 0) {
+      this.notificationStore.notifyUser({
+        type: "warning",
+        sticky: false,
+        text: _t("Match(es) cannot be replaced as they are part of a formula."),
+      });
+    } else {
+      this.notificationStore.notifyUser({
+        type: "warning",
+        sticky: false,
+        text: _t(
+          replaceableMatches +
+            " match(es) replaced. " +
+            irreplaceableMatches +
+            " match(es) cannot be replaced as they are part of a formula."
+        ),
```

Comment:
Dynamic strings can't be translated properly:

- only static strings are extracted from the code to be sent to translators
- they can't be translated as translators won't have translated every possible value of the dynamic parts. There won't be any match in the translation.

See https://www.odoo.com/documentation/17.0/developer/howtos/translations.html#variables. The doc is for python-side, but the same principles apply client-side

```suggestion
        text: _t("%(replaceable_count)s match(es) replaced. %(irreplaceable_count)s match(es) cannot be replaced as they are part of a formula.", { replaceable_count: ...,  irreplaceable_count: ...}),
```

Side note: when there are more than one placeholder, always use named placeholder (`%(replaceable_count)s`) instead of `%s`. It allows translators to reorder them if needed in some languages.